File: Server/src/main/core/network/event/MSHSWriteEvent.java
package core.network.event;

import core.cache.misc.buffer.ByteBufferUtils;
import core.network.IoSession;
import core.network.IoWriteEvent;
import core.ServerConstants;

import java.nio.ByteBuffer;

/**
 * Handles the management server handshake write event.
 * @author Emperor
 */
public final class MSHSWriteEvent extends IoWriteEvent {

	/**
	 * Constructs a new {@code MSHSWriteEvent} {@code Object}
	 * @param session The session.
	 * @param context The context.
	 */
	public MSHSWriteEvent(IoSession session, Object context) {
		super(session, context);
	}

	@Override
	public void write(IoSession session, Object context) {
		ByteBuffer	buffer = ByteBuffer.allocate(2 + ServerConstants.MS_SECRET_KEY.length());
		buffer.put((byte) 88);
		ByteBufferUtils.putString(ServerConstants.MS_SECRET_KEY, buffer);
		session.queue((ByteBuffer) buffer.flip());
	}

}

File: Server/src/main/core/network/event/MSHSWriteEvent.kt
package core.network.event

import core.ServerConstants
import core.cache.misc.buffer.ByteBufferUtils
import core.network.IoSession
import core.network.IoWriteEvent
import java.nio.ByteBuffer

/**
 * Handles the management server handshake write event.
 * @author Emperor
 */
class MSHSWriteEvent(session: IoSession, context: Any) : IoWriteEvent(session, context) {

    override fun write(session: IoSession, context: Any) {
        val buffer = ByteBuffer.allocate(2 + ServerConstants.MS_SECRET_KEY.length)
        buffer.put(88.toByte())
        ByteBufferUtils.putString(ServerConstants.MS_SECRET_KEY, buffer)
        session.queue(buffer.flip() as ByteBuffer)
    }
}